api/v1beta1/machinedeployment_webhook_test.go

```diff
@@ -393,11 +393,15 @@ func TestMachineDeploymentValidation(t *testing.T) {
 				},
 			}
 			if tt.expectErr {
-				g.Expect(md.ValidateCreate()).NotTo(Succeed())
-				g.Expect(md.ValidateUpdate(md)).NotTo(Succeed())
+				_, err := md.ValidateCreate()
```

Comment:
Not sure if it's worth validating the nil. I would expect us to add more validation once we start to return warnings.

Reply:
If we would prefer also checking for nil, I would open a follow-up help-wanted issue.

Reply:
The benefit to explicitly checking for nil now is that it sort of forces basic UT coverage for any future usages of the new warning responses (otherwise it's up to reviewers to manually ensure UT coverage).

But I agree a follow-up issue is an appropriate way of prioritizing this (i.e., low priority)

Reply:
Created an issue: https://github.com/kubernetes-sigs/cluster-api/issues/8732